### ISSUE TITLE
Remove `is_riboviz_gff`??

### ISSUE THREAD

**Issue by ewallace** [riboviz/riboviz#306]
In `bam_to_h5.R`, there's a legacy parameter `is_riboviz_gff`, (default `TRUE`), which if `FALSE` assumes fixed-length flanking regions given by the `buffer` variable. This relates to #83.

I think that we are now telling users that they have to use a riboviz-style gff, which would contain the `left_flank` and `right_flank` variables by default. So can't we remove, `is_riboviz_gff` as it will always be `TRUE`?

**Comment by mikej888** [comment]
If users must use RiboViz-style GFF then, yes, this parameter can be removed. I can do that once I've merged #315.

**Comment by mikej888** [comment]
@ewallace, in #315 you comment

> Still requires update to Remove is_riboviz_gff?? #306. I think it's possible to use just the CDS co-ordinates instead of UTR5 and UTR3 values there.

but above you comment:

> I think that we are now telling users that they have to use a riboviz-style gff, which would contain the left_flank and right_flank variables by default. So can't we remove, is_riboviz_gff as it will always be TRUE?

which seems to be contradictory (the left_flank and right_flank variables are taken from UTR5 and UTR3 values).

**Comment by ewallace** [comment]
You're right, this is contradictory from me. I have confused two issues:
1. Do we need to insist on users using a riboviz-style gff?
2. If we use a riboviz-style gff, do we need to take the flank size from the `UTR5` and UTR3 features, or could we rely only on the `CDS` feature.

For issue 2 I think it would conceptually clean up the code if only the `CDS` feature in the gff was used in `bam_to_h5.R`; that is, the `left_flank` and `right_flank` are whatever is to the left and right of the CDS respectively, rather than taking overlapping information from the UTR features, which violates the best practice that information should be stored in only one place. Also, our `check_fasta_gff` code only checks the CDS information so then we would compute on the same data that we check.

For issue 1 I'm actually not sure. Can we discuss with the team?

**Comment by mikej888** [comment]
Yes, let's discuss. I'll add 2.2 as a possible alternative release target for this.

**Comment by mikej888** [comment]
At present, for riboviz-style GFF, `left_flank` and `right_flank` are taken from UTR5 and UTR3 features in the GFF.

For `left_flank` this would be 1..`<CDS start position> - 1`.

But `right_flank` would be `<CDS end position> + 1..<?>`

The GFF doesn't have information about the CDS end position. `right_flank` is used in a number of `rtracklayer`-related calculations in the `ReadsToCountMatrix` matrix. How would it be retrieved for use in these?

The only other data available is from the SAM/BAM file but I can't see how one could get the `right_flank` value from that.

**Comment by mikej888** [comment]
If `is_riboviz_gff` is removed then `stop_in_feature` can also be removed as the latter is only used if the former is `false`.

**Comment by mikej888** [comment]
If deprecating this parameter we would need to describe the expected format of a riboviz-compliant GFF file i.e. what it should look like and what it should, or could, include.
